add cloud9 instruction in the drift-detection (#221)
aws-samples/cfn101-workshop: content/intermediate/operations/drift-detection/index.md

In the next step, you will use the AWS CloudFormation Console to create a new stack using this template:

   :::::tabs{variant="container"}
	::::tab{id="cloud9" label="Cloud9"}
	1. In the **Cloud9 terminal** navigate to `code/workspace/drift-detection`:
    :::code{language=shell showLineNumbers=false showCopyAction=true}
    cd cfn101-workshop/ccode/workspace/drift-detection
    :::
    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
    :::code{language=shell showLineNumbers=false showCopyAction=true}

>     aws cloudformation create-stack --stack-name drift-detection-workshop --template-body file://drift-detection-workshop.yaml

Reviewer: ```suggestion
    aws cloudformation create-stack --stack-name cfn-workshop-drift-detection --template-body file://drift-detection-workshop.yaml
```

    :::
    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.

>     :::code{language=shell showLineNumbers=false showCopyAction=true}

Reviewer: ```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

>     "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/drift-detection-workshop/739fafa0-e4d7-11ed-a000-12d9009553ff"

Reviewer: ```suggestion
    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-drift-detection/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

    :::
    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
    ::::
    ::::tab{id="local" label="Local Development"}
   1. Navigate to the [AWS CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
   1. From **Create stack**, choose **With new resources (standard)**.
   1. From **Specify template**, choose **Upload a template file**. Choose the `drift-detection-workshop.yaml` template mentioned earlier, and then choose **Next**.

>    1. Enter a stack name. For example, specify `drift-detection-workshop`. Choose **Next**.

Reviewer: ```suggestion
   1. Enter a stack name. For example, specify `cfn-workshop-drift-detection`. Choose **Next**.
```

   1. In **Configure stack options**, choose **Next**.

>    1. Choose **Create stack**.

Reviewer: ```suggestion
   1. Choose **Submit**.
```

1. In your favourite text editor, open the template file for the workshop.
2. Modify `Queue1`'s `MessageRetentionPeriod` to match the value shown in the **Actual** column of the drift details page you saw in the previous step. In your template, set the value for `MessageRetentionPeriod` to `172800`, that is the number of seconds in `2` days.
3. Save the template file.
1. It’s now time to update your stack! Follow steps below:

> 4. Navigate to the [CloudFormation Console](https://console.aws.amazon.com/cloudformation/).

Reviewer: ```suggestion
```

> 5. Choose your stack as before.

Reviewer: ```suggestion
```

1. It’s now time to update your stack! Follow steps below:

   :::::tabs{variant="container"}
	::::tab{id="cloud9" label="Cloud9"}

> 	1. In the **Cloud9 terminal** navigate to `code/workspace`:

Reviewer: ```suggestion
	1. In the **Cloud9 terminal** ensure your working directory is `code/workspace/drift-detection`:
```

    :::code{language=shell showLineNumbers=false showCopyAction=true}

>     cd cfn101-workshop/code/workspace

Reviewer: ```suggestion
    cd cfn101-workshop/code/workspace/drift-detection
```

    :::
    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
    :::code{language=shell showLineNumbers=false showCopyAction=true}

>     aws cloudformation update-stack --stack-name cfn-workshop-s3 --template-body file://template-and-stack.yaml

Reviewer: ```suggestion
    aws cloudformation update-stack --stack-name cfn-workshop-drift-detection --template-body file://template-and-stack.yaml
```

    :::
    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.

>     :::code{language=shell showLineNumbers=false showCopyAction=true}

Reviewer: ```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

>     :::code{language=shell showLineNumbers=false showCopyAction=true}

Reviewer: ```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

>     "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-s3/739fafa0-e4d7-11ed-a000-12d9009553ff"

Reviewer: ```suggestion
    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-drift-detection/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

    :::
    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.

>     ::::

Reviewer: Need to add in the stack drift check
```suggestion
 1. Choose the **Stack info** tab.
   1. From **Stack actions**, choose **Detect Drift**.
   1. Wait a few seconds for drift detection to complete.
   1. You should now see that the drift status is `IN_SYNC`, showing that the template and resource match.
    ::::
```

    ::::tab{id="local" label="Local Development"}
   1. Log in to the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new browser tab.
   1. Navigate to the [CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
   1. Choose your stack as before.
   1. Choose **Update**.
   1. Choose **Replace current template**, then choose **Upload a template file**.
   1. Use **Choose file** to select your updated template file.
   1. Choose **Next**.
   1. On the stack details page, choose **Next**.
   1. On the stack options page, choose **Next**.

>    1. Choose **Update Stack**.

Reviewer: ```suggestion
   1. Choose **Submit**.
```

::alert[This `UserData` script will only run the first time the instance boots. You can [create a configuration](https://aws.amazon.com/premiumsupport/knowledge-center/execute-user-data-ec2/) which will run a script on every boot, but in order to keep the template complexity low for this workshop, this template just shows simple content.]

   :::::tabs{variant="container"}
	::::tab{id="cloud9" label="Cloud9"}

> 	1. In the **Cloud9 terminal** navigate to `code/workspace`:

Reviewer: ```suggestion
	1. In the **Cloud9 terminal** ensure your working directory is `code/workspace/drift-detection`:
```

    :::code{language=shell showLineNumbers=false showCopyAction=true}

>     cd cfn101-workshop/code/workspace

Reviewer: ```suggestion
    cd cfn101-workshop/code/workspace/drift-detection
```

    :::
    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
    :::code{language=shell showLineNumbers=false showCopyAction=true}

>     aws cloudformation update-stack --stack-name cfn-workshop-s3 --template-body file://template-and-stack.yaml

Reviewer: ```suggestion
    aws cloudformation update-stack --stack-name cfn-workshop-drift-detection-challenge --template-body file://drift-detection-challenge.yaml
```

    :::
    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.

>     :::code{language=shell showLineNumbers=false showCopyAction=true}

Reviewer: ```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

>     "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-s3/739fafa0-e4d7-11ed-a000-12d9009553ff"

Reviewer: ```suggestion
    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-drift-detection-challenge/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

    :::
    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.

>     ::::

Reviewer: ```suggestion
1. Once the stack is created, select the `drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.
    ::::
```

    ::::tab{id="local" label="Local Development"}
   1. Log in to the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new browser tab.
   1. Navigate to the [AWS CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
   1. From **Create stack**, choose **With new resources (standard)**.
   1. From **Specify template**, choose **Upload a template file**, upload the `drift-detection-challenge.yaml` file and choose **Next**.

>    1. Enter a stack name, for example `drift-detection-challenge` and choose **Next**.

Reviewer: ```suggestion
   1. Enter a stack name, for example `cfn-workshop-drift-detection-challenge` and choose **Next**.
```

   1. In **Configure Stack Options**, choose **Next**.

>    1. In the next page, choose **Create stack**.

Reviewer: ```suggestion
   1. In the next page, choose **Submit**.
```

>    1. Once the stack is created, select the `drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.

Reviewer: ```suggestion
   1. Once the stack is created, select the `cfn-workshop-drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.
```

:::

Your task now is to update the stack with the new state of the resource, without causing a further interruption.

:::expand{header="Need a hint?"}
* You should detach the resource from the stack, then re-import it again with the `UserData` corrected. You do not need to convert the `UserData` to Base64.
* Refer to the [Resource Importing](/intermediate/operations/resource-importing.html) lab for more guidance.
:::
::::::expand{header="Want to see the solution?"}
1. Update the `drift-detection-challenge.yaml` template to add a `DeletionPolicy` attribute with a value of `Retain` to the `Instance1` resource. Save the file.
1. Update the stack with the updated `drift-detection-challenge.yaml` template. This tells CloudFormation that when the resource is removed from the template, it should not delete it but just stop managing it.
1. Once the stack update is complete, edit the template file again to remove the whole resource declaration (you can also choose to comment it out, using the `#` character at the start of each relevant line), and save the file.

> 1. Update the stack with the updated template file. CloudFormation will remove the instance from the stack without terminating it.

Reviewer 2: Missing Cloud 9 instructions for updating the stack

      UserData: !Base64 |
        #!/usr/bin/env bash
        echo Hello Universe
:::
1. Select the stack in the CloudFormation console, then from **Stack actions** choose **Import resources in to stack**.
1. Choose **Next**.
1. Upload the template file.
1. Enter the physical ID of the instance and choose **Next**.
1. In **Specify stack options**, choose **Next**.
1. Choose **Import resources**.

> 1. Once the stack operation is complete and the resource is imported, you can run drift detection on the stack to confirm the instance is now in sync with the stack template.

Reviewer 2: Missing Cloud 9 instructions for updating the stack
